### PR TITLE
refactor: use logger instead of `System.out.println`

### PR DESCRIPTION
**Pull request by soloturn**
would it be acceptable to you to use logger instead of system out in these cases?

### CODE REVIEW

File: engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java
Comment (BenjaminAmos) [inline engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:25]: Have we discussed about the `LOGGER` vs `logger` naming before? I personally prefer `logger`, since all capital naming tends to be reserved for constants/literals. This logger object is strictly-speaking a constant but it is used in a non-constant fashion (it performs writes, whereas normal constants only perform reads). I am aware that checkstyle probably disagrees with me on this.

File: engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java
Comment (BenjaminAmos) [inline engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: This is something that is concerning me a bit. The class defines its own `main` method, which means that it does not use the usual entrypoint. It looks like a utility program to me. Often the printed outputs of utility programs are parsed by other tools, so it may have been intentional in this rare case to not use a logger framework.

The logger framework usually prepends all outputs lines with the log level and class, which would potentially break programs that depended on the existing output format. I do not know where this tool is used, so I am unable to say definitively if this is the case.

File: engine-tests/src/test/java/org/terasology/documentation/BindingScraper.java
Comment (BenjaminAmos) [inline engine-tests/src/test/java/org/terasology/documentation/BindingScraper.java:25]: See my comment above. This also looks like a utility tool and the class-level comment strongly suggests that the output format (without a logger) is intentional.

**Review by BenjaminAmos — COMMENTED** [review]
The logger changes seem fine to me. I will leave the `LOGGER` vs `logger` bit for others to decide upon.

I am a bit more concerned about the changes to the tools in the `engine-tests/src/test/java/org/terasology/documentation` folder. I suspect that these tools may have been intentionally not using a logging framework.

File: engine-tests/src/test/java/org/terasology/documentation/BindingScraper.java
Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/BindingScraper.java:24]: hmm. you are right @BenjaminAmos . we try to get docmentation via these, i found : https://github.com/MovingBlocks/Terasology/issues/1975 ? @Cervator ?

File: engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java
Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: hmm. you are right @BenjaminAmos . we try to get docmentation via these, i found : https://github.com/MovingBlocks/Terasology/issues/1975 ? @Cervator ?

File: engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java
Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:25]: we have have both in the code base, let us quiet checkstyle.

File: engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java
Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: this occurence is for debugging, but others are not. reverted.

File: engine-tests/src/test/java/org/terasology/documentation/BindingScraper.java
Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/BindingScraper.java:25]: reverted.

File: engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java
Comment (jdrueckert) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: These changes still seem to be in... @soloturn you might want to recheck your revert

Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: rechecked. no, they are not in any more. the comments were on a line which is part of an error message, resp logging, @jdrueckert .

Comment (jdrueckert) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: maybe I misunderstood, but I thought @BenjaminAmos's point was that we shouldn't use the logger in `main` here but stick with `System.out.println`, right?
however, I still see the logger field and logger usages in this file...
@soloturn can you clarify why you say it has been reverted when it looks like it hasn't been? ö.Ö

Comment (BenjaminAmos) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: To clarify what I meant, with a logger the output might look something like this:
> 20:04:17.005 [main] WARN  o.t.engine.registry.InjectionHelper - CameraTargetSystem wanted BlockEntityRegistry injected but CoreRegistry has none.

Whereas without the logger (just using `System.out.println`) we would just get this:
> CameraTargetSystem wanted BlockEntityRegistry injected but CoreRegistry has none.

All of the parts prepended to the message might upset programs which were expecting just a plain output format. I was purely speculating, as I am unaware of where the tool might be used currently.

Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: it outputs documentation, below run ApiScraper from branch "develop". this is now println again. other things are commented to give additional hints, but messages like "processing blabla " is clearly not part of documentation, therefor commented, now it is not commented but is standard logging which can be switched on and off without changing the code, @jdrueckert .

```
00:39:15.601 [main] DEBUG org.terasology.gestalt.module.sandbox.PermissionSet -- Globally granted permission '("java.util.PropertyPermission" "reactor.schedulers.defaultBoundedElasticQueueSize" "read")'
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.terasology.engine.core.module.ModuleManager (file:/srv/http/repos/terasology/engine/build/classes/)
WARNING: Please consider reporting this to the maintainers of org.terasology.engine.core.module.ModuleManager
WARNING: System::setSecurityManager will be removed in a future release
# Modding API:

## terasology engine

* org.terasology.engine.audio (PACKAGE)
* org.terasology.engine.audio.events (PACKAGE)
* org.terasology.engine.config.Config (CLASS)
* org.terasology.engine.config.RenderingConfig (CLASS)
* org.terasology.engine.config.RenderingDebugConfig (CLASS)
* org.terasology.engine.config.facade.TelemetryConfiguration (INTERFACE)
* org.terasology.engine.config.flexible.Setting (INTERFACE)
* org.terasology.engine.config.flexible.constraints (PACKAGE)
* org.terasology.engine.context.Context (INTERFACE)
* org.terasology.engine.core.ComponentFieldUri (CLASS)
* org.terasology.engine.core.ComponentSystemManager (CLASS)
* org.terasology.engine.core.GameScheduler (CLASS)
* org.terasology.engine.core.Observer (INTERFACE)
* org.terasology.engine.core.SimpleUri (CLASS)
* org.terasology.engine.core.Time (INTERFACE)
* org.terasology.engine.core.Uri (INTERFACE)
* org.terasology.engine.core.bootstrap.ClassMetaLibrary (INTERFACE)
* org.terasology.engine.core.modes.loadProcesses.AwaitedLocalCharacterSpawnEvent (CLASS)
* org.terasology.engine.core.module.ModuleInputStream (CLASS)
* org.terasology.engine.core.module.ModuleOutputStream (CLASS)
* org.terasology.engine.core.module.SandboxFileManager (CLASS)
* org.terasology.engine.core.module.rendering (PACKAGE)

and so on and so on ...
```

Comment (jdrueckert) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: > this is now println again. other things are commented [...] now it is not commented but is standard logging

I'm so sorry, this is probably my fault for not understanding, but I still think we're not talking about the same thing.

As @BenjaminAmos said: All of the parts prepended to the message (by using output of a logger like ``Logger.info()`` instead of without a logger like ``System.out.println()``) might upset programs which were expecting just a plain output format.

Based on this, I assume we want to use ``System.out.println`` (no logger) to avoid potential issues with the parts prepended by using a logger. Do you concur with that @soloturn ?

If you don't, then I'd like to understand, why not.

If you do, I don't understand why there's still so many logger uses in the current state of the PR, see example screenshots below:
grafik
grafik

Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: @jdrueckert  you most likely executed the main method, isn't it? if you could tell me where you expect println and not a logger i'd be glad. take for example the "unknown protocol for .." message, this is not something which should go into a documentation, its just a bug or warning to be fixed.

and if we really got one wrong, unexpected, we fix it when we notice it is wrong, imo. it is just a println vs logger ...

Comment (jdrueckert) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: I didn't execute anything (yet), I only looked at the files changed and noticed more logger usages than I expected (see screenshots in my previous comment.

Comment (BenjaminAmos) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: The convention for errors is to send them to `stderr`. Normal output goes to `stdout`. Visually they go to the same console but automated programs usually only read from `stdout`.

If we can make sure the errors and warnings go to `stderr` (standard error) then it doesn't matter what format they use since they will not be parsed at all. It will also not accidentally end-up in documentation that way.

Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/ApiScraper.java:43]: it goes to stderr by default: https://www.slf4j.org/faq.html

File: engine/src/main/java/org/terasology/engine/core/PathManager.java
Comment (BenjaminAmos) [inline engine/src/main/java/org/terasology/engine/core/PathManager.java]: This comment seems to imply that not using loggers in this file was intentional? `PathManager` is one of the earliest classes called at start-up, if I recall correctly. That comment is 10 years old though (https://github.com/MovingBlocks/Terasology/commit/c65f0850f2f71c26accb1dff795735f6fb3a58e6#diff-de3478a78af92a2fe9224c0b54edc62adf1a29e08be2a922e9e0eb4766bf1e83R64).

Looking through the source files from that time, I do not see a reason why we are unable to use a logger here. Let's remove the comment to avoid any more confusion in the future.
```suggestion

```

File: engine-tests/src/test/java/org/terasology/documentation/apiScraper/ApiComparator.java
Comment (BenjaminAmos) [inline engine-tests/src/test/java/org/terasology/documentation/apiScraper/ApiComparator.java:24]: This class is another tool class and from the look of it should not be using a logger at all. Based on #3320, the output is intended to be human readable and pasted into a message, it seems.

Comment (soloturn) [reply engine-tests/src/test/java/org/terasology/documentation/apiScraper/ApiComparator.java:24]: ApiComparator takes ```API_file.txt``` as input and creates ```New_API_file.txt```, println were log messages. ```API_file.txt``` is generated by ApiSaver, same logic, no println contributing to result.

**Review by BenjaminAmos — APPROVED** [review]
The `ApiScraper` and `ApiComparator` tools were introduced in #3253. As far as I can tell, they were never used and I am unsure about how to test them directly. The output appears to be intended as more human readable than machine readable, so the logger changes may not be significantly detrimental here. Either way, merging is is unlikely to break anything, since this code appears to be unused.

The rest of the changes raise no concerns for me.

I am approving this but would still prefer a second opinion before merging.

**Pull request merged**
